odl/phantom/geometric.py

```diff
@@ -171,23 +171,345 @@ def indicate_proj_axis(space, scale_structures=0.5):
+        psi = ellip[9] * np.pi / 180
+
+        scales = [1 / a_squared, 1 / b_squared, 1 / c_squared]
+        center = (np.array([x0, y0, z0]) + 1.0) / 2.0
+
+        # Create the offset x,y and z values for the grid
+        if any([phi, theta, psi]):
+            # Rotate the points to the expected coordinate system.
+            cphi = np.cos(phi)
+            sphi = np.sin(phi)
+            ctheta = np.cos(theta)
+            stheta = np.sin(theta)
+            cpsi = np.cos(psi)
+            spsi = np.sin(psi)
+
+            mat = np.array([[cpsi * cphi - ctheta * sphi * spsi,
+                             cpsi * sphi + ctheta * cphi * spsi,
+                             spsi * stheta],
+                            [-spsi * cphi - ctheta * sphi * cpsi,
+                             -spsi * sphi + ctheta * cphi * cpsi,
+                             cpsi * stheta],
+                            [stheta * sphi,
+                             -stheta * cphi,
+                             ctheta]])
+
+            # Calculate the points that could possibly be inside the volume
+            # Since the points are rotated, we cannot do anything directional
+            # without more logic
+
+            max_radius = np.sqrt(
+                np.abs(mat).dot([a_squared, b_squared, c_squared]))
+            idx, shapes = _getshapes_3d(center, max_radius, space.shape)
+
+            subgrid = [g[idi] for g, idi in zip(grid, shapes)]
+            offset_points = [vec * (xi - x0i)[..., np.newaxis]
+                             for xi, vec, x0i in zip(subgrid,
+                                                     mat.T,
+                                                     [x0, y0, z0])]
+            rotated = offset_points[0] + offset_points[1] + offset_points[2]
+            np.square(rotated, out=rotated)
+            radius = np.dot(rotated, scales)
+        else:
+            # Calculate the points that could possibly be inside the volume
+            max_radius = np.sqrt([a_squared, b_squared, c_squared])
+            idx, shapes = _getshapes_3d(center, max_radius, space.shape)
+
+            subgrid = [g[idi] for g, idi in zip(grid, shapes)]
+            squared_dist = [ai * (xi - x0i) ** 2
+                            for xi, ai, x0i in zip(subgrid,
+                                                   scales,
+                                                   [x0, y0, z0])]
+
+            # Parentheses to get best order for broadcasting
+            radius = squared_dist[0] + (squared_dist[1] + squared_dist[2])
+
+        # Find the pixels within the ellipse
+        inside = radius <= 1
+
+        # Add the ellipse intensity to those pixels
+        p[idx][inside] += intensity
+
+    return space.element(p)
+
+
+def ellipse_phantom(space, ellipses):
+    """Return a phantom given by ellipses.
+
+    Parameters
+    ----------
+    space : `DiscreteLp`
+        Space the phantom should live in, must be 2 or 3 dimensional.
```

Comment:
"live in" weird, better "be created in"
The whole sentence could be clearer, like "Space in which the phantom is created, must be 2- or 3-dimensional."